ksqldb-execution/src/test/java/io/confluent/ksql/execution/codegen/helpers/SearchedCaseFunctionTest.java

```diff
@@ -185,20 +187,23 @@ public void shouldWorkForStructValues() {
   public void shouldReturnDefaultIfNoMatch() {
     // Given:
     final List<SearchedCaseFunction.LazyWhenClause<Integer>> lazyWhenClauses = ImmutableList.of(
-        SearchedCaseFunction.whenClause(() -> false, () -> 1),
-        SearchedCaseFunction.whenClause(() -> false, () -> 2),
-        SearchedCaseFunction.whenClause(() -> false, () -> 3),
-        SearchedCaseFunction.whenClause(() -> false, () -> 4)
+        SearchedCaseFunction.whenClause(() -> true, () -> null)
     );
 
     // When:
     final Integer result = SearchedCaseFunction.searchedCaseFunction(
         lazyWhenClauses,
-        () -> 10
+        () -> {
+          throw new AssertionError("Should not be called");
+        }
     );
 
     // Then:
-    assertThat(result, equalTo(10));
+    assertThat(result, is(nullValue()));
+  }
+
+  @Test
+  public void shouldHandleNullReturnValues() {
```

Comment:
I think you probably meant to keep the previous test and add one here?

Reply:
OOps